pkg/endpointmanager/manager.go

```diff
@@ -330,14 +333,27 @@ func (mgr *endpointManager) LookupIP(ip netip.Addr) (ep *endpoint.Endpoint) {
 	return ep
 }
 
-// LookupPodName looks up endpoint by namespace + pod name
-func (mgr *endpointManager) LookupPodName(name string) *endpoint.Endpoint {
+// LookupCEPName looks up an endpoint by its K8s namespace + cep name
+func (mgr *endpointManager) LookupCEPName(namespacedName string) *endpoint.Endpoint {
 	mgr.mutex.RLock()
-	ep := mgr.lookupPodNameLocked(name)
+	ep := mgr.lookupCEPNameLocked(namespacedName)
 	mgr.mutex.RUnlock()
 	return ep
 }
 
+// GetEndpointsByPodName looks up endpoints by namespace + pod name
+func (mgr *endpointManager) GetEndpointsByPodName(namespacedName string) []*endpoint.Endpoint {
+	mgr.mutex.RLock()
+	eps := make([]*endpoint.Endpoint, 0, 1)
+	for _, ep := range mgr.endpoints {
```

Comment:
That is a valid concern. The current indexing unfortunately is not powerful enough to avoid that. I looked into it, and we would have to introduce another `endpointsByPodName map[string]map[uint16]*endpoint.Endpoint` (or similar) index in the endpoint manager to fix it, which seems a bit intrusive. This could be a great use-case for statedb.

Given that the number of endpoints on a node should be in the order of few hundred pods at most I'm not too worried yet.